site-service/src/test/java/uk/ac/ox/ndph/mts/site_service/repository/HapiFhirRepositoryTests.java

```diff
@@ -0,0 +1,163 @@
+package uk.ac.ox.ndph.mts.site_service.repository;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Organization;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.ResearchStudy;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import uk.ac.ox.ndph.mts.site_service.exception.RestException;
+
+@ExtendWith(MockitoExtension.class)
+class HapiFhirRepositoryTests {
+
+    @Mock
+    private FhirContextWrapper fhirContextWrapper;
+
+    @Captor
+    private ArgumentCaptor<Bundle> bundleCaptor;
+
+    @Test
+    void TestHapiRepository_WhenSaveSite_SendsBundleWithTrasactionType()
```

Comment:
This is probably SaveOrganization not site, right?

Reply:
This has been fixed